tools/reactive-controllers/src/LanguageResolution.ts

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2022 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import type { ReactiveController, ReactiveElement } from 'lit';
+import { ProvideLang } from '@spectrum-web-components/theme';
+
+export class LanguageResolutionControler implements ReactiveController {
+    host: ReactiveElement;
+    language!: string;
```

Comment:
Since this is public, it may be safer to type it as nullable, since it starts undefined. That way, a consumer querying `this.languageController.language` will know to handle the null-or-undefined case.

```suggestion
    language?: string;
```